### PR TITLE
Fix index calculations in GlobStaticMem

### PR DESCRIPTION
**Pull request by fmoessbauer**
@ddiefenthaler, @devreal  This PR adds a unit test for issue #532

### CODE REVIEW

**Comment by devreal** [comment]
Thanks! Should we use this PR to fix the issue? Otherwise we'd break CI everywhere else...

**Comment by fmoessbauer** [comment]
@devreal: just to fix the issue and when fixed, we can merge it. @ddiefenthaler told me yesterday, that he will have a look at this today.

**Comment by devreal** [comment]
Great! @ddiefenthaler please let us know if you need help with this!

**Comment by fmoessbauer** [comment]
At first: @fuchsto and @rkowalewski  own me a beer!

While the fix is quite minimal, it revealed some conceptual problems as described in issue #539.

File: dash/include/dash/memory/GlobStaticMem.h [hunk 1]
```diff
@@ -523,9 +524,10 @@ class GlobStaticMem
     DASH_LOG_TRACE_VAR("GlobStaticMem.at", lunit);
     // Apply global unit to global pointer:
     dart_gptr_setunit(&gptr, lunit);
+    // increment locally only
+    gptr.addr_or_offs.offset += local_index * sizeof(value_type);
     // Apply local offset to global pointer:
     pointer res_gptr(*this, gptr);
-    res_gptr += local_index;
```
Comment (devreal) [inline dash/include/dash/memory/GlobStaticMem.h:528]: Why does this not work? Shouldn't it be the same as incrementing the offset if the `dart_gptr_t` by `local_index * sizeof(value_type)`?

Comment (fmoessbauer) [reply dash/include/dash/memory/GlobStaticMem.h:528]: The difference is that the dart version only increments locally while the dash version first checks if the increment is past the local address range. As the local size of the memory on unit 1 is 96 (vs 144, unit 0) the dash version belives that a increment of 128 is a pointer to index 32 on the next unit. The actual bug are the capacity issues as stated in the linked issue.

Comment (devreal) [reply dash/include/dash/memory/GlobStaticMem.h:528]: Ahh, gotcha! I agree that it seems more efficient to operate on the `dart_gptr_t` directly, anyway. Thanks for fixing this!

**Review by devreal — APPROVED** [review]

**Pull request merged**